homeassistant/components/overkiz/alarm_control_panel.py

```diff
@@ -0,0 +1,304 @@
+}
+
+
+def _state_myfox_alarm_controller(
+    select_state: Callable[[str], OverkizStateType]
+) -> str:
+    """Return the state of the device."""
+    if (
+        cast(str, select_state(OverkizState.CORE_INTRUSION))
+        == OverkizCommandParam.DETECTED
+    ):
+        return STATE_ALARM_TRIGGERED
+
+    return MAP_MYFOX_STATUS_STATE[
+        cast(str, select_state(OverkizState.MYFOX_ALARM_STATUS))
+    ]
+
+
+MAP_ARM_TYPE: dict[str, str] = {
+    OverkizCommandParam.DISARMED: STATE_ALARM_DISARMED,
+    OverkizCommandParam.ARMED_DAY: STATE_ALARM_ARMED_HOME,
+    OverkizCommandParam.ARMED_NIGHT: STATE_ALARM_ARMED_NIGHT,
+    OverkizCommandParam.ARMED: STATE_ALARM_ARMED_AWAY,
+}
+
+
+def _state_alarm_panel_controller(
+    select_state: Callable[[str], OverkizStateType]
+) -> str:
+    """Return the state of the device."""
+
+    return MAP_ARM_TYPE[
+        cast(str, select_state(OverkizState.VERISURE_ALARM_PANEL_MAIN_ARM_TYPE))
+    ]
+
+
+ALARM_DESCRIPTIONS: list[OverkizAlarmDescription] = [
+    # TSKAlarmController
+    OverkizAlarmDescription(
+        key=UIWidget.TSKALARM_CONTROLLER,
+        entity_registry_enabled_default=False,
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY
+            | SUPPORT_ALARM_ARM_HOME
+            | SUPPORT_ALARM_ARM_NIGHT
+            | SUPPORT_ALARM_TRIGGER
+        ),
+        fn_state=_state_tsk_alarm_controller,
+        alarm_disarm=lambda execute_command: execute_command(OverkizCommand.ALARM_OFF),
+        alarm_arm_home=lambda execute_command: execute_command(
+            OverkizCommand.SET_TARGET_ALARM_MODE, OverkizCommandParam.PARTIAL_1
+        ),
+        alarm_arm_night=lambda execute_command: execute_command(
+            OverkizCommand.SET_TARGET_ALARM_MODE, OverkizCommandParam.PARTIAL_2
+        ),
+        alarm_arm_away=lambda execute_command: execute_command(
+            OverkizCommand.SET_TARGET_ALARM_MODE, OverkizCommandParam.TOTAL
+        ),
+        alarm_trigger=lambda execute_command: execute_command(OverkizCommand.ALARM_ON),
+    ),
+    # StatefulAlarmController
+    OverkizAlarmDescription(
+        key=UIWidget.STATEFUL_ALARM_CONTROLLER,
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_NIGHT
+        ),
+        fn_state=_state_stateful_alarm_controller,
+        alarm_disarm=lambda execute_command: execute_command(OverkizCommand.DISARM),
+        alarm_arm_home=lambda execute_command: execute_command(
+            OverkizCommand.ALARM_ZONE_ON, [OverkizCommandParam.A]
+        ),
+        alarm_arm_night=lambda execute_command: execute_command(
+            OverkizCommand.ALARM_ZONE_ON, [OverkizCommandParam.A, OverkizCommandParam.B]
+        ),
+        alarm_arm_away=lambda execute_command: execute_command(
+            OverkizCommand.ALARM_ZONE_ON,
+            [OverkizCommandParam.A, OverkizCommandParam.B, OverkizCommandParam.C],
+        ),
```

Comment:
WIth the above amend, it becomes:
```suggestion
        alarm_disarm=OverkizCommand.DISARM,
        alarm_arm_home=OverkizCommand.ALARM_ZONE_ON,
        alarm_arm_home_args=[OverkizCommandParam.A],
        alarm_arm_night=OverkizCommand.ALARM_ZONE_ON,
        alarm_arm_night_args=[OverkizCommandParam.A, OverkizCommandParam.B],
        alarm_arm_away=OverkizCommand.ALARM_ZONE_ON,
        alarm_arm_away_args=[
            OverkizCommandParam.A,
            OverkizCommandParam.B,
            OverkizCommandParam.C,
        ],
```